Feature/query microservices

jwomeara opened the pull request.

ivakegg committed:
Merge branch 'integration' into feature/queryMicroservices

jwomeara committed:
Modified a couple of things and fixed a few tests with the integration merge.

jwomeara committed:
pulling in latest audit changes

jwomeara committed:
pulled in latest audit updates

jwomeara committed:
Fixed metrics version in docker compose

jwomeara committed:
Updated accumulo utils and datawave starter versions.

jwomeara committed:
Updated the audit api and audit starter versions

jwomeara committed:
Updated authorization api version

jwomeara committed:
Updated accumulo api version.

jwomeara committed:
Updated config service with latest changes

jwomeara committed:
updated dictionary and accumulo utils version

jwomeara committed:
Updated hazelcast version

jwomeara committed:
bumped query metric api version

jwomeara committed:
Updated pki for query microservices docker deployment.

jwomeara committed:
Fixing the docker compose bootstrap to avoid duplicate extra hosts issue.

jwomeara committed:
Merge remote-tracking branch 'origin/integration' into feature/queryMicroservices

jwomeara committed:
adding latest datawave changes

jwomeara committed:
updated metadata utils

jwomeara committed:
updated docker compose to use the latest version of each service.

jwomeara committed:
Added a cached results spring boot starter and updated associated classes accordingly.

jwomeara committed:
Merge remote-tracking branch 'origin/feature/spring-boot-2.x-accumulo-2.1' into feature/queryMicroservicesAccumulo2.1

jwomeara committed:
fixing spotbugs error

jwomeara committed:
fixed install-ingest.sh for datawave.

jwomeara committed:
Updates to get bulk results map reduce query working again.

jwomeara committed:
Merge remote-tracking branch 'origin/integration' into feature/queryMicroservicesAccumulo2.1

jwomeara committed:
Merge remote-tracking branch 'origin/release/version5.1' into feature/queryMicroservicesAccumulo2.1

jwomeara committed:
formatting

ivakegg committed:
Updates per a race condition discovered in the kafka variant of the
query executor tests.

jwomeara committed:
Update README.md

ivakegg committed:
accumulo service merge from main

jwomeara committed:
Merge remote-tracking branch 'origin/integration' into feature/queryMicroservicesAccumulo2.1

jwomeara committed:
fixed a test that broke after merging integration

jwomeara committed:
Merge remote-tracking branch 'origin/feature/queryMicroservicesAccumulo2.1' into feature/queryMicroservicesAccumulo2.1

jwomeara committed:
Merge remote-tracking branch 'origin/feature/spring-boot-2.x-accumulo-2.1' into feature/queryMicroservicesAccumulo2.1

jwomeara committed:
Added logic from Datawave Principal Login Module to the datawave spring boot starter.

jwomeara committed:
Merge remote-tracking branch 'origin/integration' into feature/queryMicroservicesAccumulo2.1

jwomeara committed:
Updated datawave to 5.4.0-SNAPSHOT

jwomeara committed:
Updated ProxiedUserDetails to use dynamic type for newInstance method.

jwomeara committed:
Merge remote-tracking branch 'origin/integration' into feature/queryMicroservicesAccumulo2.1

jwomeara committed:
updated audit service to stop intermittent test failures.

jwomeara committed:
Merge remote-tracking branch 'origin/feature/spring-boot-2.x-accumulo-2.1' into feature/queryMicroservicesAccumulo2.1

jwomeara committed:
Updated the query results publisher and setup the kafka manager to create topics manually.

jwomeara committed:
Merge remote-tracking branch 'origin/integration' into feature/queryMicroservicesAccumulo2.1

jwomeara committed:
Added github actions for some submodules

jwomeara committed:
Merge remote-tracking branch 'origin/integration' into feature/queryMicroservicesAccumulo2.1

jwomeara committed:
Merge remote-tracking branch 'origin/feature/spring-boot-2.x-accumulo-2.1' into feature/queryMicroservicesAccumulo2.1

jwomeara committed:
Merge branch 'feature/spring-boot-2.x-accumulo-2.1' into feature/queryMicroservicesAccumulo2.1

jwomeara committed:
Updated github actions to avoid running out of disk space.

jwomeara committed:
Merge remote-tracking branch 'origin/feature/spring-boot-2.x-accumulo-2.1' into feature/queryMicroservicesAccumulo2.1

jwomeara committed:
Merge branch 'integration' into feature/queryMicroservicesAccumulo2.1

jwomeara committed:
Merge branch 'feature/spring-boot-2.x-accumulo-2.1' into feature/queryMicroservicesAccumulo2.1

jwomeara committed:
Updated import sorting for microservices using impsort plugin.

jwomeara committed:
Merge branch 'feature/spring-boot-2.x-accumulo-2.1' into feature/queryMicroservicesAccumulo2.1

jwomeara committed:
Merge branch 'feature/spring-boot-2.x-accumulo-2.1' into feature/queryMicroservicesAccumulo2.1

jwomeara committed:
Merge branch 'feature/spring-boot-2.x-accumulo-2.1' into feature/queryMicroservicesAccumulo2.1

jwomeara committed:
Merge remote-tracking branch 'origin/feature/spring-boot-2.x-accumulo-2.1' into feature/queryMicroservicesAccumulo2.1

jwomeara committed:
fixing merge conflicts

jwomeara committed:
Merge branch 'feature/spring-boot-2.x-accumulo-2.1' into feature/queryMicroservicesAccumulo2.1

jwomeara committed:
updated metadata utils

jwomeara committed:
Merge branch 'integration' into feature/queryMicroservicesAccumulo2.1

jwomeara committed:
Merge remote-tracking branch 'origin/integration' into feature/queryMicroservicesAccumulo2.1

jwomeara committed:
Moved Query, QueryImpl to query-api

jwomeara committed:
Removed usage of zookeeper stringutils

jwomeara committed:
added package-info.java for Query, QueryImpl

jwomeara committed:
updated modification service logic to use ProxiedUserDetails.

jwomeara committed:
Removed unnecessary jackson annotations to address enunciate issues

jwomeara committed:
Merge remote-tracking branch 'origin/feature/spring-boot-2.x-accumulo-2.1' into feature/queryMicroservicesAccumulo2.1

jwomeara committed:
Updated profile ids for submodules

jwomeara committed:
changes to get the github actions working again

jwomeara committed:
Merge remote-tracking branch 'origin/integration' into feature/queryMicroservicesAccumulo2.1

jwomeara committed:
attempting to fix github actions

jwomeara committed:
typo

jwomeara committed:
udpated kafka config

jwomeara committed:
Merge remote-tracking branch 'origin/feature/spring-boot-2.x-accumulo-2.1' into feature/queryMicroservicesAccumulo2.1

jwomeara committed:
Merge branch 'feature/spring-boot-2.x-accumulo-2.1' into feature/queryMicroservicesAccumulo2.1

jwomeara committed:
removed banned dependency

jwomeara committed:
bumped release version

jwomeara committed:
pulled in latest updates from spring boot 2.x

jwomeara committed:
formatting

jwomeara committed:
Merge remote-tracking branch 'origin/feature/spring-boot-2.x-accumulo-2.1' into feature/queryMicroservicesAccumulo2.1

jwomeara committed:
fixing docker compose kafka setup

jwomeara committed:
Merge remote-tracking branch 'origin/feature/spring-boot-2.x-accumulo-2.1' into feature/queryMicroservicesAccumulo2.1

jwomeara committed:
updated deploy poms to work with the latest version of maven

jwomeara committed:
Merge branch 'feature/spring-boot-2.x-accumulo-2.1' into feature/queryMicroservicesAccumulo2.1

jwomeara committed:
bumped version for some modules

jwomeara committed:
bumped microservice service parent version

jwomeara committed:
Merge branch 'integration' into feature/queryMicroservicesAccumulo2.1

jwomeara committed:
Updated with latest changes from main/integration

jwomeara committed:
fixed the composite query logic for the query microservices

jwomeara committed:
Merge remote-tracking branch 'origin/integration' into feature/queryMicroservicesAccumulo2.1

jwomeara committed:
Merge remote-tracking branch 'origin/integration' into feature/queryMicroservicesAccumulo2.1

jwomeara committed:
updated submodules

ivakegg committed:
Added v1 to the model path

jwomeara committed:
Merge remote-tracking branch 'origin/integration' into feature/queryMicroservicesAccumulo2.1

jwomeara committed:
For this branch we need to checkout all submodules when building tests.

jwomeara committed:
Added alerman's run cache settings for docker

jwomeara committed:
Updated github actions to build query microservices correctly

jwomeara committed:
fixed build cache config typo

jwomeara committed:
Merge remote-tracking branch 'origin/integration' into feature/queryMicroservicesAccumulo2.1

jwomeara committed:
Updated query service to use pool header, and allow for admin override via parameter

jwomeara committed:
Updated github ci to make sure it runs against the queryMicroservices branch

jwomeara committed:
Updated lookup uuid to accept multiple terms per field

jwomeara committed:
Updated confirm ack configuration for query metric service, and added configuration to make rest metrics updates possible

jwomeara committed:
Merge branch 'integration' into feature/queryMicroservices

jwomeara committed:
Merge remote-tracking branch 'origin/integration' into feature/queryMicroservices

ivakegg committed:
Added support for long running queries

jwomeara committed:
Added configcheck command line utility

jwomeara committed:
Merge remote-tracking branch 'origin/feature/queryMicroservices' into feature/queryMicroservices

jwomeara committed:
updated next call logic for long running queries

jwomeara committed:
Merge remote-tracking branch 'origin/integration' into feature/queryMicroservices

jwomeara committed:
updated query metric service and starter submodules

jwomeara committed:
Added some plugins to the config check service

jwomeara committed:
Updated configcheck tests to trim whitespace

jwomeara committed:
fixing unit tests for config check

ivakegg committed:
Changed from max long running query timeouts to long running query
timeout

ivakegg committed:
Merge commit '8b17cfec11da84e5ec594d71751f3784095d9256' into feature/queryMicroservices

jwomeara committed:
Updated the query starter submodule

jwomeara committed:
Merge remote-tracking branch 'origin/integration' into feature/queryMicroservices

jwomeara committed:
Merge remote-tracking branch 'origin/integration' into feature/queryMicroservices

jwomeara committed:
Merge remote-tracking branch 'origin/integration' into feature/queryMicroservices

ivakegg committed:
Merge remote-tracking branch 'origin/integration' into feature/queryMicroservices

jwomeara committed:
Merge remote-tracking branch 'origin/integration' into feature/queryMicroservices

jwomeara committed:
Merge remote-tracking branch 'origin/feature/queryMicroservices' into feature/queryMicroservices

jwomeara committed:
updated submodules

jwomeara committed:
Merge remote-tracking branch 'origin/integration' into feature/queryMicroservices

jwomeara committed:
Merge remote-tracking branch 'origin/integration' into feature/queryMicroservices

jwomeara committed:
Merge remote-tracking branch 'origin/integration' into feature/queryMicroservices

jwomeara committed:
fixed composite query logic test

jwomeara committed:
Merge remote-tracking branch 'origin/integration' into feature/queryMicroservices

jwomeara committed:
Refactored query scripts to use common code

jwomeara committed:
Merge branch 'integration' into feature/queryMicroservices

jwomeara committed:
Merge remote-tracking branch 'origin/integration' into feature/queryMicroservices

jwomeara committed:
Updated quickstart docker-maven-plugin to not cleanup old tagged releases

jwomeara committed:
updated common-utils submodule

jwomeara committed:
Merge remote-tracking branch 'origin/integration' into feature/queryMicroservices

jwomeara committed:
Updated LookupService to behave the same as the webservice

jwomeara committed:
Merge remote-tracking branch 'origin/integration' into feature/queryMicroservices

jwomeara committed:
Fixed TypedValue (de)serialization, and added a result post processor to handle things like counts

jwomeara committed:
reset docker-compose quickstart to just be accumulo, not web

jwomeara committed:
Changed how the result postprocessor runs to support result aggregation for long running queries

jwomeara committed:
Renamed aggregateResults to reduceResults

jwomeara committed:
Merge remote-tracking branch 'origin/integration' into feature/queryMicroservices

jwomeara committed:
Merge remote-tracking branch 'origin/integration' into feature/queryMicroservices

jwomeara committed:
Merge remote-tracking branch 'origin/integration' into feature/queryMicroservices

jwomeara committed:
Merge remote-tracking branch 'origin/integration' into feature/queryMicroservices

jwomeara committed:
Merge remote-tracking branch 'origin/integration' into feature/queryMicroservices

jwomeara committed:
Merge remote-tracking branch 'origin/integration' into feature/queryMicroservices

jwomeara committed:
Added junit-jupiter engine to datawave core pom

jwomeara committed:
Updated query logic factory to add logic map

jwomeara committed:
Updated EdgeQueryLogicFactory configuration

jwomeara committed:
Removed unused RemoteQueryLogic and cleaned up RemoteEventQueryLogic class definition:

jwomeara committed:
Merge remote-tracking branch 'origin/integration' into feature/queryMicroservices

jwomeara committed:
Implemented authorization and query federation for the query microservices

jwomeara committed:
Fixed copy constructor parameter assignment in QueryData

jwomeara committed:
Merge remote-tracking branch 'origin/integration' into feature/queryMicroservices

Commit:
Formatting job fix

jwomeara committed:
Fixing build/formatter issues for latest changes pulled in from integration

jwomeara committed:
Merge remote-tracking branch 'origin/feature/queryMicroservices' into feature/queryMicroservices

jwomeara committed:
Fixed SSDeepIngestQueryTest

jwomeara committed:
Updated usage of remote user operations for query microservices

jwomeara committed:
Fixed QueryPlan constructor

jwomeara committed:
Moved filtering and delegating query logic functionality into datawave-core-query so it can be used by the microservices

jwomeara committed:
Merge remote-tracking branch 'origin/integration' into feature/queryMicroservices

jwomeara committed:
formatting

jwomeara committed:
Updated submodules

jwomeara committed:
Updated package names for commons.lang3 classes due to type-utils fix

jwomeara committed:
Moved the AuthorizationsPredicate class to authorization-api

jwomeara committed:
Removed hadoop dependencies from pom

jwomeara committed:
Updated edge configuration to reset param variables before loading new values

jwomeara committed:
Merge remote-tracking branch 'origin/integration' into feature/queryMicroservices

jwomeara committed:
Excluded log4j from our maven plugins

jwomeara committed:
Merge remote-tracking branch 'origin/integration' into feature/queryMicroservices

jwomeara committed:
Merge remote-tracking branch 'origin/integration' into feature/queryMicroservices

ivakegg committed:
Merge remote-tracking branch 'origin/integration' into feature/queryMicroservices

jwomeara committed:
Updated in-memory-accumulo javadoc plugin to exclude log4j

jwomeara committed:
Updated microservice-parent maven-dependency-plugin to exclude log4j

jwomeara committed:
Removed infinispan from query starter

jwomeara committed:
Remove validation call from LoadModel

jwomeara committed:
fixed formatting

jwomeara committed:
Updated datawave ingest jar libs

jwomeara committed:
Merge remote-tracking branch 'origin/integration' into feature/queryMicroservices

billoley committed:
QueryMetricQueryLogic must copy runAsMetricsAdministrator setting in copy constructor

jwomeara committed:
Revert query metric query logic back to it's original state

jwomeara committed:
Merge remote-tracking branch 'origin/integration' into feature/queryMicroservices

foster33 commented on contrib/datawave-quickstart/bin/modification.sh:
```diff
@@ -0,0 +1,295 @@
+
+function modificationHelp() {
+    echo
+    echo " The $( printGreen "datawaveModification" ) shell function allows you submit modification requestson demand to DataWave's"
```
```suggestion
    echo " The $( printGreen "datawaveModification" ) shell function allows you submit modification requests on demand to DataWave's"
```
Minor typo

alerman commented on .github/workflows/tests.yml:
```diff
@@ -73,55 +76,57 @@ jobs:
-  #           || { echo "***** TESTS FAILED. Attempting retry."; $RUN_TESTS; } \
-  #           || { echo "***** TESTS FAILED. Attempting final retry."; $RUN_TESTS; }
+  # Build the code and run the microservice unit/integration tests.
+  build-and-test-microservices:
```
Is the intention to leave this uncommented moving forward?

jwomeara replied:
This is going to have to go away actually.  I had something similar for the spring-boot-2.x branch, and it's a nice idea, but unless everything is a snapshot version it doesn't really add much value.  I would prefer to comment it out AFTER this is merged just to make sure we are running this github action right up until the last possible second.

alerman commented on .github/workflows/tests.yml:
```diff
@@ -142,14 +147,92 @@ jobs:
         TAG=$(mvn -s $GITHUB_WORKSPACE/.github/workflows/settings.xml -q -N -Dmaven.build.cache.enabled=false -Dexec.executable='echo' -Dexec.args='${project.version}' exec:exec)
         contrib/datawave-quickstart/docker/docker-build.sh ${TAG} --docker-opts "${DOCKER_BUILD_OPTS}"
 
+  compose-build-and-test:
```
Is the plan to require this step to pass on all branch builds? Should this be a MR only check or otherwise restricted or do we believe it needs to be run on all commits?

jwomeara replied:
this should go away AFTER the PR is merged.  I want to keep it in there until that point though just to make sure any changes to the microservices are tested appropriately.

alerman commented on .github/workflows/tests.yml:
```diff
@@ -142,14 +147,92 @@ jobs:
+      run: |
+        mvn -s $GITHUB_WORKSPACE/.github/workflows/settings.xml -B -V -e -Pcompose -Dmicroservice-docker -Dquickstart-docker -Ddeploy -Dtar -DskipTests clean install
+        
+        # free up some space so that we don't run out
```
Is there some minimum amount of space we need that we could check for, then perform cleanup if necessary?

jwomeara replied:
This is an issue I ran into when building all of the docker images.  I don't think this will be an issue once this github action is removed after merging.

alerman commented on contrib/datawave-quickstart/bin/services/accumulo/bootstrap.sh:
```diff
@@ -42,6 +42,7 @@ syncLimit=5
 clientPort=2181
 dataDir=${DW_CLOUD_DATA}/zookeeper
 maxClientCnxns=100
+4lw.commands.whitelist=ruok,wchs
```
Any reason not to use the wildcard (*) to enable all here?

jwomeara replied:
I don't even remember why this was added or whether it's still needed.  @ivakegg?

alerman replied:
In one of the versions of ZK they made them all disabled by default. I am sure that at least ruok had to be added for health checks, I just don't know why we would limit it and not just enable all.

jwomeara commented on .mvn/maven-build-cache-config.xml:
```diff
@@ -71,6 +71,22 @@
                         <goal>install</goal>
                     </goals>
                 </goalsList>
+                <goalsList artifactId="docker-maven-plugin">
```
@alerman I'm not sure any of these changes actually help.  I still have issues with the build cache when building everything.

jwomeara commented on contrib/datawave-quickstart/bin/services/datawave/install-ingest.sh:
```diff
@@ -82,6 +82,8 @@ else
    if [ -d ${DW_DATAWAVE_INGEST_HOME}/accumulo-warehouse/lib/ext ]; then
       cp ${DW_DATAWAVE_INGEST_HOME}/accumulo-warehouse/lib/ext/*.jar ${ACCUMULO_HOME}/lib/ext > /dev/null 2>&1
    fi
+   info "Removing bundled guava jar from ${ACCUMULO_HOME}/lib"
```
I'm not sure this is necessary anymore.  I'll look into it.

foster33 commented on core/connection-pool/src/main/java/datawave/core/query/runner/AccumuloConnectionRequestMap.java:
```diff
@@ -0,0 +1,105 @@
+    private Map<String,List<Pair<Map<String,String>,Thread>>> getConnectionThreadMap = new HashMap<>();
+
+    public boolean cancelConnectionRequest(String id, String userDn) {
+        // this call checks that the Principal used for the connection request and th connection cancel are the same
```
```suggestion
        // this call checks that the Principal used for the connection request and the connection cancel are the same
```
Typo

foster33 commented on core/map-reduce/src/main/java/datawave/core/mapreduce/bulkresults/map/ApplicationContextAwareMapper.java:
```diff
@@ -0,0 +1,76 @@
+
+    private static Logger log = Logger.getLogger(ApplicationContextAwareMapper.class);
+
+    //
```
```suggestion
```
empty comment

foster33 commented on core/query/src/main/java/datawave/core/query/logic/QueryLogic.java:
```diff
@@ -285,23 +296,23 @@ default String getResponseClass(Query query) throws QueryException {
 
     /**
-     * Check that the user has one of the required roles principal my be null when there is no intent to control access to QueryLogic
+     * Check that the user has one of the required roles. userRoles my be null when there is no intent to control access to QueryLogic
```
```suggestion
     * Check that the user has one of the required roles. userRoles may be null when there is no intent to control access to QueryLogic
```

foster33 commented on core/query/src/main/java/datawave/core/query/logic/composite/CompositeQueryLogic.java:
```diff
@@ -598,21 +599,83 @@ public Set<String> getExampleQueries() {
+        CheckpointableQueryLogic logic = (CheckpointableQueryLogic) queryLogics.get(compositeCheckpoint.getDelegateQueryLogic());
+        if (logic == null) {
+            throw new UnsupportedOperationException(
+                            "Cannot update query checkpoint because delegate query logic [" + compositeCheckpoint.getDelegateQueryLogic() + "]does not exist");
```
```suggestion
                            "Cannot update query checkpoint because delegate query logic [" + compositeCheckpoint.getDelegateQueryLogic() + "] does not exist");
```

foster33 commented on core/query/src/main/java/datawave/core/query/logic/composite/CompositeQueryLogic.java:
```diff
@@ -598,21 +599,83 @@ public Set<String> getExampleQueries() {
+        CheckpointableQueryLogic logic = (CheckpointableQueryLogic) queryLogics.get(compositeCheckpoint.getDelegateQueryLogic());
+        if (logic == null) {
+            throw new UnsupportedOperationException(
+                            "Cannot update query checkpoint because delegate query logic [" + compositeCheckpoint.getDelegateQueryLogic() + "]does not exist");
```
```suggestion
                            "Cannot update query checkpoint because delegate query logic [" + compositeCheckpoint.getDelegateQueryLogic() + "] does not exist");
```

foster33 commented on warehouse/query-core/src/main/java/datawave/query/scheduler/Scheduler.java:
```diff
@@ -1,24 +1,27 @@
-
-public abstract class Scheduler implements CloseableIterable<Entry<Key,Value>> {
 
+/**
+ *
+ */
```
empty

foster33 commented on warehouse/query-core/src/main/java/datawave/query/tables/BatchScannerSession.java:
```diff
@@ -40,18 +40,32 @@
-import datawave.webservice.query.Query;
 
-public class BatchScannerSession extends ScannerSession implements Iterator<Entry<Key,Value>>, FutureCallback<Scan>, SessionArbiter, UncaughtExceptionHandler {
+/**
+ *
+ */
```
empty

jwomeara commented on .github/workflows/tests.yml:
```diff
@@ -73,55 +76,57 @@ jobs:
     steps:
     - name: Checkout Code
       uses: actions/checkout@v3
+      with:
+        submodules: 'recursive'
```
This is useful while this exists as a PR, but I will remove this once the PR is merged.

jwomeara commented on .github/workflows/tests.yml:
```diff
@@ -142,14 +147,92 @@ jobs:
         TAG=$(mvn -s $GITHUB_WORKSPACE/.github/workflows/settings.xml -q -N -Dmaven.build.cache.enabled=false -Dexec.executable='echo' -Dexec.args='${project.version}' exec:exec)
         contrib/datawave-quickstart/docker/docker-build.sh ${TAG} --docker-opts "${DOCKER_BUILD_OPTS}"
 
+  compose-build-and-test:
```
This is useful while this is a PR, but I will remove this once it's merged.

jwomeara commented on .gitignore:
```diff
@@ -52,3 +52,6 @@ services/sample_configuration/*-dev.yml
 /.dockerignore
 /.maven-dockerignore
+
+/.dockerignore
```
i'll remove these dupes.

jwomeara commented on .github/workflows/tests.yml:
```diff
@@ -56,6 +57,8 @@ jobs:
     steps:
     - name: Checkout Code
       uses: actions/checkout@v3
+      with:
```
This is useful while this exists as a PR, but I will remove this once the PR is merged.

jwomeara committed:
pr feedback

jwomeara committed:
PR feedback

jwomeara commented on core/connection-pool/src/main/java/datawave/core/common/result/ConnectionFactoryResponse.java:
```diff
@@ -77,7 +77,7 @@ public String getMainContent() {
 
         builder.append("\n");
 
-        builder.append("table.connectionRequests td, table.connectionPools th {");
```
@ivakegg is this correct?

ivakegg replied:
reverted

jwomeara commented on core/query/src/main/java/datawave/core/query/configuration/GenericQueryConfiguration.java:
```diff
@@ -29,13 +32,19 @@
-public class GenericQueryConfiguration {
+public class GenericQueryConfiguration implements Serializable {
+    // is this execution expected to be checkpointable (changes how we allocate ranges to scanners)
+    private boolean checkpointable = false;
```
is a check for `this instanceof CheckpointableQueryConfiguration` not suitable enough for this?

apmoriarty replied:
My preference is for instanceOf, it's more reliable than a boolean that could be accidentally set by a non-checkpointable query logic

jwomeara replied:
the only downside with removing this is that you wouldn't be able to force the query to run in an uncheckpointable way without the boolean.  not sure if we care about that at all.  @ivakegg?

FineAndDandy replied:
there could definitely be a case for a best effort no checkpoint query

ivakegg replied:
So the point of this is that an implementation may support checkpointing, however we may not want to run it in a checkpointable way for latency purposes (think UUID lookup which uses the shard query logic and the shard query configuration)

jwomeara replied:
If you update the other class to be `enableCheckpoints` or `checkpointsEnabled` can you do the same here?

jwomeara commented on core/query/src/main/java/datawave/core/query/configuration/GenericQueryConfiguration.java:
```diff
@@ -219,10 +291,32 @@ public boolean canRunQuery() {
+        if (o == null || getClass() != o.getClass())
+            return false;
+        GenericQueryConfiguration that = (GenericQueryConfiguration) o;
+        return isCheckpointable() == that.isCheckpointable() && getBaseIteratorPriority() == that.getBaseIteratorPriority()
```
Probably need to make sure all of the class members are accounted for in equals and hashcode since this branch has been around as long as it has.

alerman replied:
For other projects I have used project lombok's @EqualsAndHashCode. Matbe we consider that? https://projectlombok.org/features/EqualsAndHashCode

apmoriarty replied:
Another alternative is Apache's EqualsBuilder and HashCodeBuilder, we've used those elsewhere in the project.

jwomeara replied:
@alerman unfortunately, for some of these classes i think there are class members which we *don't* want to consider when comparing for equality.

jwomeara replied:
After looking at this, I think all of the members are accounted for in both equals and hashCode.  I don't have a strong opinion about which method to use, other than to say that what's there is already implemented...  🤷

alerman commented on warehouse/core/src/main/java/datawave/ingest/util/cache/Loader.java:
```diff
@@ -70,6 +70,7 @@ public ListenableFuture<V> reload(final K key, final V oldValue) {
                 log.trace("Reloading synchronously");
             try {
                 build(null);
+
```
Adding this new line intentional?

jwomeara replied:
Doing a blame on this line in intellij shows that this was part of the original code commit...

alerman commented on warehouse/ingest-configuration/src/main/resources/config/myjson-ingest-config.xml:
```diff
@@ -208,11 +208,11 @@
-        <!--<value>PREMIERED</value>-->
-        <!--<description>Known date field to be used, if found, for the shard row id. Otherwise, current date will be used</description>-->
-    <!--</property>-->
+    <property>
```
Any Idea why this had been commented out? Seems like we would want this on in general?

jwomeara replied:
I think I talked with @keith-ratcliffe at the time.  I think I had to do this in order to make the date ranges for my queries work reliably?  it's been a while, so im not sure.

alerman commented on warehouse/query-core/.attach_pid18134:
inadvertant add?

alerman commented on core/connection-pool/src/main/java/datawave/core/common/cache/SharedCacheCoordinator.java:
```diff
@@ -98,12 +91,7 @@ public interface EvictionCallback {
-                    @ConfigProperty(name = "dw.cacheCoordinator.numLocks", defaultValue = "300") int numLocks,
-                    @ConfigProperty(name = "dw.cacheCoordinator.maxRetries", defaultValue = "10") int maxRetries) {
+    public SharedCacheCoordinator(String namespace, String zookeeperConnectionString, int evictionReaperIntervalInSeconds, int numLocks, int maxRetries) {
         ArgumentChecker.notNull(namespace, zookeeperConnectionString);
```
Do we have a dependency on Guava? If so, any reason not to use Preconditions.checkNotNull rather than implementing ourselves?

jwomeara replied:
The notNull check wasn't added by me.  I don't really have an opinion here - I just left it as-is.

alerman commented on core/connection-pool/src/test/java/datawave/core/common/cache/SharedCacheCoordinatorTest.java:
```diff
@@ -181,7 +182,7 @@ public void stateChanged(CuratorFramework client, ConnectionState newState) {}
             testingZooKeeperServer.kill();
 
-            for (int i = 0; i < 15; ++i) {
+            for (int i = 0; i < 20; ++i) {
```
Why 20? Are we just arbitrarily guessing how many iterations to do?

jwomeara replied:
This is a timing thing.  Bumping the limit up increases the amount of time to perform the operation.

alerman commented on core/connection-pool/src/test/resources/log4j.properties:
```diff
@@ -0,0 +1,6 @@
+log4j.rootCategory=INFO, CONSOLE
+
```
Not sure if this is a log4j that would get passed through into production, but consider adding the monitorInterval setting to allow updates without app restart.

    monitorInterval = 30

jwomeara replied:
This is the test configuration.  As a rule I try not to deploy test code to production.  ;)

alerman commented on core/query/src/main/java/datawave/core/query/logic/QueryKey.java:
```diff
@@ -0,0 +1,104 @@
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public class QueryKey implements Serializable {
+    private static final long serialVersionUID = -2589618312956104322L;
```
Curious how you decided on this serialVersionUID? Same question for other places. If you used some method to generate, we should be explicit about that wherever it is used so that future devs know how to update it.

jwomeara replied:
I have never understood how these are generated, or even what their usefulness is.  @ivakegg?

FineAndDandy replied:
the serialVersionUID has to be assigned for any Serializable class for compatibility. This value is likely the value automatically generated by the compiler, but it can also be versioned manually for compatibility

ivakegg replied:
https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/io/Serializable.html

jwomeara commented on core/query/src/main/java/datawave/core/query/logic/CheckpointableQueryLogic.java:
```diff
@@ -0,0 +1,60 @@
+     *
+     * @return true if checkpointable
+     */
+    boolean isCheckpointable();
```
@ivakegg this has always struck me as a little odd.  If the query logic implements CheckpointableQueryLogic, isn't that sign enough that the query logic is checkpointable?  Are you using this as more of an on/off switch?

apmoriarty replied:
I agree. The only use I can see for this is if you wanted to do A/B testing with checkpointable logic disabled.

jwomeara replied:
I think I could get on board for renaming this to something along the lines of 'enableCheckpoints' or something like that.  This is less about whether the query logic is checkpointable, and more about whether we want to use it in a checkpointable fashion.

ivakegg replied:
So the setCheckpointable method below would be the method to change to enableCheckpoints if you wish.  Its the CreateTask/NextTask classes used by the query executor service that needs to know whether the query can be checkpointed of not.  If not, then in the create task all of the results are pulled at that point.  If is it then other executor service instances can handle pulling results as needed.  This mechanism allows us to migrate query logics into the query microservices framework but still support those query logics that have not been migrated.

jwomeara replied:
I would prefer enableCheckpoints or checkpointsEnabled, but that's not a hill I'm willing to die on.

alerman commented on docker/README.md:
```diff
@@ -111,7 +163,7 @@ Docker Compose version v2.21.0
 #### Datawave Quickstart
 
-Prior to starting docker compose, you need to use the Datawave Quickstart to deploy working instances of Hadoop, Zookeeper, and Accumulo, along with some sample datasets for query.
+Prior to starting docker compose, you need to use the Qatawave Quickstart to deploy working instances of Hadoop, Zookeeper, and Accumulo, along with some sample datasets for query.
```
Typo QataWave. 

Obviously, it should be QuataWave

alerman commented on web-services/atom/src/main/java/datawave/webservice/atom/AtomServiceBean.java:
```diff
@@ -120,8 +120,12 @@ public void setup() {}
         Set<Authorizations> auths = new HashSet<>();
+        String userDN = null;
+        Collection<String> proxyServers = null;
         if (p instanceof DatawavePrincipal) {
```
This code is repeated in this class. Create a method to get the DN and server from the DP object perhaps?

jwomeara replied:
i could go either way on this one...

alerman commented on warehouse/query-core/src/main/java/datawave/query/language/functions/lucene/LuceneQueryFunction.java:
```diff
@@ -8,7 +8,6 @@
 import datawave.query.search.WildcardFieldedFilter;
 
-@Deprecated
 public abstract class LuceneQueryFunction implements QueryFunction {
```
Are we undeprecating this?

jwomeara replied:
i think i questioned whether these were actually unused or not.  i'll add the deprecation back in.

jwomeara commented on microservices/README.md:
```diff
@@ -0,0 +1,95 @@
+# DATAWAVE Query Microservices
```
Probably need to review this for accuracy.

jwomeara replied:
Actually, I'm going to get rid of the entire thing since it was more of a place to take notes during early development.

alerman commented on web-services/common/src/main/java/datawave/webservice/common/audit/AuditBean.java:
```diff
@@ -14,6 +14,7 @@
 
 import org.apache.log4j.Logger;
 import org.jboss.resteasy.annotations.GZIP;
+import org.springframework.util.MultiValueMap;
```
Whats the difference betweein this and MultivaluedMap? Can we use all of one or the other? Im worried about this causing confusion for developers down the line.

alerman replied:
Same comment for other classes in this package.

jwomeara replied:
MultivaluedMap is a jboss/CDI/wildfly construct, so it should only really ever have been used in the webservice code.  MultiValueMap is the spring boot version of that class.

alerman reviewed (commented):
Few small comments, but over all it's good.

jwomeara commented on warehouse/query-core/src/main/java/datawave/query/config/ShardQueryConfiguration.java:
```diff
@@ -713,6 +723,69 @@ public ShardQueryConfiguration(ShardQueryConfiguration other) {
+     * @param other
+     * @param queries
+     */
+    public ShardQueryConfiguration(ShardQueryConfiguration other, Collection<QueryData> queries) {
```
This probably needs to be updated.

jwomeara replied:
Looking at this a little closer, I guess it's ok.  The full copy constructor can manage copying all of the fields, while this one only handles the ones that are important for a checkpoint.  @ivakegg how did you decide on the list of members that need to be copied to support checkpointable queries, and could these members be moved to the CheckpointableQueryConfiguration class (or something equivalent) to better distinguish between properties that are important to checkpointable queries, and properties that aren't?

ivakegg replied:
There may be a few of the properties that are common to all checkpointable queries (i.e. need to consider other query logics like the edge queries for example).  I will try to assess the commonalities.

ivakegg replied:
The common fields required are already covered by the GenericQueryConfiguration copy constructor.  I am cleaning out any redundencies however.

ivakegg replied:
Found some redundency with the super copy constructor and some incorrectly overridden Query members.

jwomeara commented on warehouse/query-core/src/main/java/datawave/query/config/ShardQueryConfiguration.java:
```diff
@@ -2698,4 +2753,160 @@ public boolean isSortQueryByCounts() {
+        if (!super.equals(o))
+            return false;
+        ShardQueryConfiguration that = (ShardQueryConfiguration) o;
+        return isTldQuery() == that.isTldQuery() && isDisableIndexOnlyDocuments() == that.isDisableIndexOnlyDocuments()
```
this and the hash method probably need to be updated.

jwomeara commented on warehouse/query-core/src/main/java/datawave/query/iterator/profile/EvaluationTrackingNestedIterator.java:
```diff
@@ -25,6 +25,7 @@ public EvaluationTrackingNestedIterator(QuerySpan.Stage stageName, QuerySpan myS
 
     @Override
     public T next() {
+        mySpan.next();
```
@ivakegg just checking if this is still needed.  not sure what the context surrounding this change was.

ivakegg replied:
I am pretty sure this is still correct.  This is tracking the next/seek calls in the evaluation pipeline.

jwomeara replied:
ok, cool.  I just wasn't familiar with it so wanted to bring it to your attention.

jwomeara commented on warehouse/query-core/src/main/java/datawave/query/tables/RemoteEventQueryLogic.java:
```diff
@@ -69,16 +68,19 @@ public RemoteEventQueryLogic clone() {
 
     @Override
     public Set<String> getOptionalQueryParameters() {
+        // @TODO Get an instanceof the underlying query logic?
```
@ivakegg is this still a TODO?

ivakegg replied:
I don't think so.  I will remove in the main line.

ivakegg replied:
This was already removed in the main line.  Perhaps a bad merge somewhere along the line.  I will remove it here.

ivakegg replied:
done

jwomeara replied:
cool, thanks.  i did many, many, MANY merges so no surprise to me if i merged something in an undesirable way.

jwomeara commented on warehouse/query-core/src/main/java/datawave/query/tables/shard/CountResultPostprocessor.java:
```diff
@@ -0,0 +1,100 @@
+import datawave.webservice.query.result.event.EventBase;
+import datawave.webservice.query.result.event.FieldBase;
+
+public class CountResultPostprocessor implements ResultPostprocessor {
```
@ivakegg I think we need result postprocessors for other things as well, like group by queries.

ivakegg replied:
Do we not use the DocumentTransformer mechanism?  The GroupingTransform and UniqueTransform should be incorporated into that.  What we may be missing is the persistence of the grouping results or the unique bloom filter since we no longer have a stateful RunningQuery mechanism anymore.  I will check into that.  This is on the list of things to check in the ticket.

jwomeara replied:
By the time the results get to the query service they are already transformed results, instead of individual events/documents that can be aggregated by the document transformer.  So we need a way to aggregate the results in that format, which is what the result postprocessor does.  

I already added functionality to persist the results in the queue for queries that need some form of final aggregation (e.g. count queries).  To see how that works, check out NextCall.java and search for `returnIntermediateResults`.  Essentially, if this is a query that needs to be aggregated fully before it can be returned, and the next call times out or short circuits or whatever, the aggregated results seen so far will be published back into the queue.  

This is where results are published back into the queue.
https://github.com/NationalSecurityAgency/datawave-query-service/blob/feature/queryMicroservices/service/src/main/java/datawave/microservice/query/runner/NextCall.java#L158

There is also a new boolean in the GenericQueryConfiguration (reduceResults) which is used to indicate that results need to be reduced before anything is returned to the user.

jwomeara commented on web-services/query/src/main/java/datawave/webservice/query/runner/QueryExecutorBean.java:
```diff
@@ -437,7 +466,7 @@ private QueryData validateQuery(String queryLogicName, MultivaluedMap<String,Str
         log.debug(queryParameters);
         qp.clear();
-        qp.setRequestHeaders(httpHeaders != null ? httpHeaders.getRequestHeaders() : null);
+        qp.setRequestHeaders(httpHeaders != null ? MapUtils.toMultiValueMap(httpHeaders.getRequestHeaders()) : null);
```
I'm not happy with this MapUtils stuff.  I will take another look at that.

jwomeara replied:
ehhhh, after looking at the usages of this class (MapUtils) I do think it's still necessary...

jwomeara committed:
PR Feedback

foster33 commented on .gitmodules:
This is missing a module for query-storage. For example, when `git submodule update --init --recursive` is ran, a fatal error occurs and outputs ```fatal: No url found for submodule path 'microservices/services/query-storage' in .gitmodules```

jwomeara replied:
got it.  i intended to remove it.  i'll handle that properly with the next commit.

jwomeara replied:
let me know if that's working for you now.

foster33 replied:
It is good now

PaulsGitHubs replied:
I'm actually so relieved it's working.

jwomeara committed:
PR feedback

jwomeara committed:
PR Feedback

jwomeara committed:
Merge remote-tracking branch 'origin/integration' into feature/queryMicroservices

jwomeara committed:
Merge remote-tracking branch 'origin/integration' into feature/queryMicroservices

jwomeara committed:
Merge remote-tracking branch 'origin/integration' into feature/queryMicroservices

jwomeara committed:
PR feedback

jwomeara committed:
PR feedback

jwomeara committed:
PR Feedback

jwomeara committed:
PR Feedback

jwomeara committed:
PR feedback

jwomeara committed:
PR feedback

jwomeara committed:
PR feedback

jwomeara committed:
pr feedback

jwomeara committed:
PR feedback

jwomeara committed:
pr feedback

jwomeara committed:
pr feedback

jwomeara committed:
pr feedback

jwomeara committed:
pr feedback

jwomeara committed:
Merge remote-tracking branch 'origin/integration' into feature/queryMicroservices

FineAndDandy commented on core/common-util/src/main/java/datawave/core/query/cache/ResultsPage.java:
```diff
@@ -0,0 +1,45 @@
+/**
+ *
+ */
+public class ResultsPage<T> {
```
is a ResultsPage intended to be a one to one to a query or one to many?

jwomeara replied:
I don't understand your question.  ResultsPage isn't a new concept - it's currently in use on integration.

FineAndDandy commented on core/connection-pool/src/main/java/datawave/core/common/connection/AccumuloConnectionFactoryImpl.java:
```diff
@@ -0,0 +1,381 @@
+        this.pools = new HashMap<>();
+
+        if (this.connectionPoolsConfiguration == null) {
+            log.error("connectionPoolsConfiguration was null - aborting init()");
```
should this throw IllegalStateException?

jwomeara replied:
Good question.  This is identical to what is currently in use on integration though.

FineAndDandy commented on core/connection-pool/src/main/java/datawave/core/query/runner/AccumuloConnectionRequestMap.java:
```diff
@@ -0,0 +1,105 @@
+    /**
+     * This maps the query-id to a pair containing the tracking map (see the AccumuloConnectionFactory) and the thread handling the request
+     */
+    private Map<String,List<Pair<Map<String,String>,Thread>>> getConnectionThreadMap = new HashMap<>();
```
```suggestion
    private Map<String,List<Pair<Map<String,String>,Thread>>> connectionThreadMap = new HashMap<>();
```

jwomeara replied:
yuck.  good catch.

FineAndDandy commented on core/modification/src/main/java/datawave/modification/MutableMetadataUUIDHandler.java:
```diff
@@ -322,13 +329,7 @@ else if (f.getName().equalsIgnoreCase(event.getIdType()) && fieldCount < 1 && co
                                         log.trace("Submitting request to MutableMetadataHandler from MutableMetadataUUIDHandler: " + modReq);
 
                                     // make sure user isn't null or empty
```
this is no longer being checked add the null/empty check or remove the comment

jwomeara replied:
on it.

FineAndDandy commented on core/modification/src/main/java/datawave/modification/cache/ModificationCache.java:
```diff
@@ -0,0 +1,120 @@
+            throw new RuntimeException("Error during initialization of ModificationCacheBean", e);
+        } finally {
+            if (null != s)
+                s.close();
```
if s.close() can throw an exception it would be possible to leak the connection

jwomeara replied:
FWIW this code was lifted, unchanged from ModificationCacheBean.  This was just an attempt to strip the core functionality from the webservices module, and put it in a common place.

FineAndDandy commented on core/modification/src/main/java/datawave/modification/configuration/ModificationServiceConfiguration.java:
```diff
@@ -47,14 +48,18 @@ public void setSecurityMarkingExemptFields(List<String> securityMarkingExemptFie
-    public QueryExecutorBean getQueryService() {
-        return queryService;
+    public ModificationQueryService getQueryService(ProxiedUserDetails userDetails) {
+        return queryServiceFactory.createService(userDetails);
```
potential npe

jwomeara replied:
FWIW a similar problem exists on integration.  On integration we just grab the query service and start using it.  What would you like to see done here?  Personally, if the factory is null I would like to see it blow up so that we know we aren't configured correctly.

FineAndDandy replied:
throw an IllegalStateException?

jwomeara replied:
Both of those are RuntimeExceptions...  If it's all the same to you, I would rather just stick with the NPE.

FineAndDandy commented on core/query/src/main/java/datawave/core/query/configuration/QueryData.java:
```diff
@@ -0,0 +1,576 @@
+    }
+
+    public void setTableName(String tableName) {
+        this.tableName = tableName;
```
resetHashCode()

jwomeara replied:
this `resetHashCode()` stuff feels really hokey, but ok.

FineAndDandy commented on core/query/src/main/java/datawave/core/query/configuration/QueryData.java:
```diff
@@ -0,0 +1,576 @@
+        }
+
+        private IteratorSetting getIteratorSetting(JsonNode node) throws JsonProcessingException {
+            IteratorSetting setting = new IteratorSetting(1, "a", "a");
```
this initialization is a bit strange without a comment

jwomeara replied:
I agree.  @ivakegg what is the rhyme or reason for these constructor params?

ivakegg replied:
reworked to avoid hokey constructor params

jwomeara committed:
pr feedback

jwomeara committed:
pr feedback

jwomeara committed:
Merge remote-tracking branch 'origin/integration' into feature/queryMicroservices

jwomeara committed:
PR feedback

FineAndDandy commented on core/query/src/main/java/datawave/core/query/logic/DelegatingQueryLogic.java:
```diff
@@ -353,6 +323,51 @@ public void validate(Map<String,List<String>> parameters) throws IllegalArgument
+    @Override
+    public void setMaxConcurrentTasks(int maxConcurrentTasks) {
+        delegate.setMaxConcurrentTasks(maxConcurrentTasks);
+    }
+
+    @Override
+    public boolean canRunQuery(Collection<String> userRoles) {
+        return delegate.canRunQuery(userRoles);
+    }
+
+    @Override
+    public void setRequiredRoles(Set<String> requiredRoles) {
+        delegate.setRequiredRoles(requiredRoles);
+    }
+
+    @Override
+    public Set<String> getRequiredRoles() {
+        return delegate.getRequiredRoles();
+    }
+
+    @Override
+    public ProxiedUserDetails getCurrentUser() {
+        return delegate.getCurrentUser();
+    }
+
+    @Override
+    public void setCurrentUser(ProxiedUserDetails currentUser) {
+        delegate.setCurrentUser(currentUser);
+    }
+
+    @Override
+    public ProxiedUserDetails getServerUser() {
+        return delegate.getServerUser();
+    }
+
+    @Override
+    public void setServerUser(ProxiedUserDetails serverUser) {
+        delegate.setServerUser(serverUser);
+    }
+
```
after seeing identical code in the BaseQueryLogic maybe this should be pulled out into a separate class and composed?

jwomeara replied:
I don't understand your comment.  BaseQueryLogic isn't doing any kind of delegation like this class is doing.

FineAndDandy commented on core/query/src/main/java/datawave/core/query/logic/composite/CompositeQueryLogic.java:
```diff
@@ -241,9 +234,9 @@ public Set<Authorizations> updateRuntimeAuthorizationsAndQueryAuths(QueryLogic<?
+        Set<Authorizations> downgradedAuths = AuthorizationsUtil.getDowngradedAuthorizations(validQueryAuthorizations, currentUser, queryUser);
         if (log.isTraceEnabled()) {
-            log.trace("Principal auths for user " + principal.getPrimaryUser().getCommonName() + " are " + principal.getPrimaryUser().getAuths());
+            log.trace("Principal auths for user " + currentUser.getPrimaryUser().getCommonName() + " are " + currentUser.getPrimaryUser().getAuths());
```
can these names be changed or javadoced to reflect what they are intended to represent?

jwomeara replied:
Which names?  Changed to what?

jwomeara committed:
PR feedback

jwomeara committed:
pr feedback

jwomeara committed:
pr feedback

ivakegg committed:
Updated in-memory-accumulo commit

ivakegg committed:
in-memory-accumulo

jwomeara committed:
formatting

jwomeara committed:
Merge remote-tracking branch 'origin/feature/queryMicroservices' into feature/queryMicroservices

jwomeara committed:
Merge remote-tracking branch 'origin/integration' into feature/queryMicroservices

ivakegg committed:
query starter

jwomeara committed:
pr feedback

jwomeara committed:
Merge remote-tracking branch 'origin/feature/queryMicroservices' into feature/queryMicroservices

jwomeara committed:
pr feedback

jwomeara committed:
pr feedback

jwomeara committed:
pr feedback

jwomeara committed:
pr feedback

jwomeara committed:
pr feedback

jwomeara committed:
pr feedback

jwomeara committed:
Merge remote-tracking branch 'origin/integration' into feature/queryMicroservices

ivakegg committed:
Remoted CachedQueryStatus mechanism

jwomeara committed:
Merge remote-tracking branch 'origin/integration' into feature/queryMicroservices

jwomeara committed:
pr feedback

jwomeara committed:
pr feedback

jwomeara committed:
pr feedback

jwomeara committed:
Merge remote-tracking branch 'origin/integration' into feature/queryMicroservices

ivakegg committed:
pr feedback

ivakegg committed:
pr updates

jwomeara committed:
pr feedback

jwomeara committed:
Merge remote-tracking branch 'origin/feature/queryMicroservices' into feature/queryMicroservices

jwomeara committed:
Merge remote-tracking branch 'origin/integration' into feature/queryMicroservices

ivakegg committed:
pr review: found duplicate properties between configurations and the
super GenericQueryConfiguration class.  Removed redundant copies in
ShardQueryConfiguration checkpoint copy constructor

ivakegg committed:
Merge branch 'feature/queryMicroservices' of github.com:NationalSecurityAgency/datawave into feature/queryMicroservices

ivakegg committed:
review comments

ivakegg committed:
review comments

ivakegg committed:
Reverted path changes in dictionary service

ivakegg committed:
pr review updates

billoley commented on web-services/query/src/test/java/datawave/webservice/query/runner/QueryExecutorBeanTest.java:
```diff
@@ -145,7 +146,7 @@ public class QueryExecutorBeanTest {
     private QueryMetricsBean metrics;
     private QueryLogicFactoryImpl queryLogicFactory;
-    private QueryExpirationConfiguration queryExpirationConf;
+    private QueryExpirationProperties queryExpirationConf;
```
Maybe: queryExpirationProperties

jwomeara replied:
I was just trying to keep the name consistent with what it has always been.  Do you have a strong opinion about changing this?

billoley commented on web-services/query/src/test/java/datawave/webservice/query/runner/ExtendedQueryExecutorBeanTest.java:
```diff
@@ -242,10 +240,10 @@ public void setupBefore() throws Exception {
-        queryExpirationConf.setPageSizeShortCircuitCheckTime(45);
-        queryExpirationConf.setPageShortCircuitTimeout(58);
-        queryExpirationConf.setCallTime(60);
+        queryExpirationConf = new QueryExpirationProperties();
```
queryExpirationProperties

billoley commented on web-services/query/src/test/java/datawave/webservice/query/cache/RunningQueryTimingImplTest.java:
QueryExpirationProperties conf -->  QueryExpirationProperties queryExpirationProperties (multiple occurrences)

billoley commented on web-services/query/src/test/java/datawave/webservice/query/cache/QueryExpirationBeanTest.java:
```diff
@@ -70,11 +71,11 @@ public void testRemoveIdleOrExpired() throws Exception {
-        QueryExpirationConfiguration expirationConfiguration = new QueryExpirationConfiguration();
-        setInternalState(expirationConfiguration, "idleTimeMinutes", expireTime);
-        setInternalState(expirationConfiguration, "callTimeMinutes", expireTime);
+        QueryExpirationProperties expirationConfiguration = new QueryExpirationProperties();
```
queryExpirationProperties

billoley commented on web-services/query/src/main/java/datawave/webservice/query/cache/QueryExpirationBean.java:
```diff
@@ -44,7 +45,7 @@ public class QueryExpirationBean {
     @Inject
     @SpringBean(refreshable = true)
-    private QueryExpirationConfiguration conf;
+    private QueryExpirationProperties conf;
```
queryExpirationProperties

billoley commented on warehouse/query-core/src/main/java/datawave/query/tables/edge/EdgeQueryLogic.java:
Not the only place that this changes, but I really like how you got rid of the dupication of settings by using the configuration objects

jwomeara committed:
Merge remote-tracking branch 'origin/integration' into feature/queryMicroservices

jwomeara committed:
updated executor to use QueryStatusUpdateUtil

jwomeara committed:
updated query microservice submodule

jwomeara committed:
make github action tests match integration

jwomeara committed:
Merge remote-tracking branch 'origin/integration' into feature/queryMicroservices
